### DISCUSSION TITLE
Drop Item hotkey

### DISCUSSION THREAD

**Discussion by Petethegoat** [FakeFishGames/Barotrauma#10122] [Code questions and modding support]
**update:**
Made a PR: #10133
Added one handed item dropping too. 

**update:**
Fixed being able to drop handcuffs, and added the ability to drop items into containers. This works with two handed and single items, and works nicely with loaders, lockers, and presumably more. Probably some edge cases there that can be addressed, but notably you can't use it to drop items into other people's inventories at the moment.
Also input masking from GUIs stops it too, which is annoying for some things like reactors in particular.

**original post**
Similar suggestion to #7444, but with implementation and questions.

I wanted to be able to drop diving suits and railgun shells without having to reach all the way over and click and drag.

I have implemented a drop hotkey here: https://github.com/Regalis11/Barotrauma/compare/master...Petethegoat:Barotrauma:master

You could expand this to single handed items but obviously then the question is should you drop one first, or both at once?
Personally I don't find it cumbersome for most items as I can unequip them with the keyboard already, so I didn't try to answer that, and it only works items that fill both hands.

As far as the code, I haven't gotten very familiar yet so I don't know if I'm really checking in the right place or really doing much of any appropriately. It does work, but any feedback or pointers would be appreciated.

Secondly, to show up nicely in the keybinding menu it needs translations, and I'm not sure where to add those in git. Easy to add them locally, and it works fine in English at least. But if I need to add that to the PR I'll need some guidance.

https://user-images.githubusercontent.com/734762/196785619-c442a590-6017-4cf3-8762-fa42d9f1f8ac.mp4

https://user-images.githubusercontent.com/734762/196813831-3f39bea1-5712-4e9d-92df-d89d0bf451cc.mp4

**Comment by MaaiG** [comment]
> I wanted to be able to drop diving suits and railgun shells without having to reach all the way over and click and drag.

No, I don’t agree )) I think taking off the suit (and railgun shells) requires a lot of effort 😬! You can’t do it without a mouse 😉)).
Small objects can be moved easily, while large ones require extra effort...
Donning a diving suit is a whole process!

> You could expand this to single handed items but obviously then the question is should you drop one first, or both at once?

I think drop one first!
The only question is, which one?
Maybe PKM and LKM can be responsible for different hands...

I have no way of checking, but I have a question.
Where will the item be moved if the cabinet is linked to a deconstructor or fabricator? To the deconstructor or cabinet?
The one I’m standing next to?

**Reply by Petethegoat** [reply]
> Where will the item be moved if the cabinet is linked to a deconstructor or fabricator? To the deconstructor or cabinet?

Good question. It goes into the primary thing that the player interacted with. If you click the deconstructor, it'll drop items into that until it's full, and then onto the floor.
